### PR TITLE
Flintlock rilfe to musket name change and addition of a proper rifle

### LINKED ISSUE #37204: Flintlock rifles are misnamed.
**Is your feature request related to a problem? Please describe.**  
What the game currently calls a flintlock rifle is clearly supposed to be a musket, given it's accuracy and (lack of) range (Also see #37203 on the matter of range, as they should have more dispersion and more range), that is a smoothbore weapon, not a rifled one. 

**Describe the solution you'd like**  
The name of the weapon should be changed to "Flintlock Musket".

**Describe alternatives you've considered**  
Changing the stats (more accuracy, a _lot_ more reload time) to actually represent a flintlock rifle and adding an actual flintlock musket as a different item. Or first changing the name of the current item and then creating a new item to represent a flintlock musket.

**Additional context**  
https://en.wikipedia.org/wiki/Brown_Bess

https://en.wikipedia.org/wiki/Long_rifle

### PR DESCRIPTION
**Pull request by Ker-Nes**
#### Summary
```SUMMARY: Content "Name change and an addition to flintlock weaponry"```

#### Purpose of change
Close #37204 
Related #37203 

#### Describe the solution
This changes the name of the flintlock rifle to flintlock musket, and add a proper flintlock rifle with longer reload time and less dispersion. Still needs to be further balanced alongside the rest of the flintlock weapons, but decided to first add the new gun and change the name of the old one, as this one was easier and faster.
For the moment, the new flintlock rifle has a much longer reload time, and bit more range and half the dispersion. It should not unbalance anything, as even with very low skill, even a flintlock carbine still manages good accuracy at longer ranges than their actual max range in game.
Also, still need to properly populate spawn lists so it appears as loot in the appropriate places, but given how much rarer actual flintlock rifles were compared to smoothbore muskets, I feel that deserves further discussion first.
If possible, the flintlock rifle should also probably foul even faster than the flintlock musket, but I'm unsure how to change that.

#### Describe alternatives you've considered
First rebalancing the gun ranges before changing names and adding the new gun.
Changing the name and description of the flintlock carbine to be a musket and leaving the old flintlock rifle as is. 

#### Testing
Edited JSON files on an already compiled game, game loads as normal, old guns had their name changed, new ones were presenting in the crafting list.

#### Additional context
https://en.wikipedia.org/wiki/Brown_Bess#Variations For the archetypal musket, as well as it's variants, including the much shorter cavalry carbine model.
https://en.wikipedia.org/wiki/Long_rifle for probably the closest thing to an archetypal muzzle loading flintlock rifle
https://en.wikipedia.org/wiki/M1819_Hall_rifle For inspiration about what might be interesting to add later, specifically early breechloading weapons which in some periods still used flintlock and paper cartridges, allowing reloading faster than even most muskets, but still giving rifle accuracy.

### CODE REVIEW

**Comment by foulman** [comment]
I'm glad I saw this, I was just starting to do something related for Magiclysm. I'll use the longrifle stats instead, as that is actually what I thought the flintlock was. Good job!

**Comment by thethunderhawk** [comment]
We really need Minie Balls! There’s no reason a survivor with the know-how to build a rifle from scratch and rifle it themselves wouldn’t want to try to make their own Minie Balls for better reload speed and reliability. Designed properly they shouldn’t take much more time to load than a solid ball in a smoothbore musket.

**Comment by Ker-Nes** [comment]
On minie balls, as I understand, it might be better to create a new item to represent that type of weapon, with a recipe similar to the .36 and .44 paper cartridges, as well as a new gun to represent the later rifled muskets of that saw great use in the civil war, as opposed to the earlier "Brown Bess" type of smoothbore muskets, and (somewhat rare and quite prone to fouling) rifled flintlocks.

**Comment by Ker-Nes** [comment]
As a sidenote, I am considering requiring a gunsmith kit for the rifle (for the rifling button), although I could believe that an early (and rather primitive) version of rifling, such as the one in the flintlock rifles could be done with the tools already used by the flintlock weapons recipe. Probably dropping the sturdy tag too.

**Comment by foulman** [comment]
> 
> 
> On minie balls, as I understand, it might be better to create a new item to represent that type of weapon, with a recipe similar to the .36 and .44 paper cartridges, as well as a new gun to represent the later rifled muskets of that saw great use in the civil war, as opposed to the earlier "Brown Bess" type of smoothbore muskets, and (somewhat rare and quite prone to fouling) rifled flintlocks.
> 
> Edit:
> @foulman please keep in mind that the current flintlocks also need a change in stats, at least in my opinion, as mentioned in the other issue up there. They have too little dispersion, instead by limited by max range, which I feel to be the exactly opposite of the ideal way of representing these weapons.

@Ker-Nes Well technically I'm only copying some of the base stats of the weapon and making "techno-magic" versions that don't shoot actual projectiles, (in this case I'm calling it a "flamelock" and it shoots fire) it's more of a retro-style choice, so it's not a big deal, but if I see any huge adjustments to things like weight/volume I'll keep an eye out for it and take it into account.

File: data/json/items/gun/flintlock.json [hunk 1]
```diff
@@ -97,5 +97,14 @@
       [ "stock mount", 1 ]
     ],
     "flags": [ "NEVER_JAMS", "DURABLE_MELEE" ]
+  },
+  {
+    "id": "longrifle_flintlock",
+    "copy-from": "rifle_flintlock",
+    "type": "GUN",
+    "name": "flintlock rifle",
+    "description": "Also called a longrifle, this rifled flintlock gun has signficantly increased accuracy at the cost of taking even longer to reload.",
```
Comment (mlangsdorf) [inline data/json/items/gun/flintlock.json]: I think "Kentucky rifle" or "Pennsylvania rifle" are more common names for this gun in modern America, and a little more distinct than longrifle.

Comment (Ker-Nes) [reply data/json/items/gun/flintlock.json]: Sounds good. Which of the two do you think more common for the general region the game is assumed to take place in?

Comment (anothersimulacrum) [reply data/json/items/gun/flintlock.json]: Pennsylvania is closer to New England, maybe that?

Comment (Ker-Nes) [reply data/json/items/gun/flintlock.json]: > Pennsylvania is closer to New England, maybe that?

It's what I'm, thinking, but I don't know if it's one of those things where they call it "Pennsylvania rifle" in Kentucky and "Kentucky rifle" in Pennsylvania, so if someone who's better versed in historical trivia of the American Northeast, I'd appreciate it. If no one is sure, I will just use Pennsylvania for the moment, and change later if anyone has further information.

Comment (Ker-Nes) [reply data/json/items/gun/flintlock.json]: And changed the description to "also called a Pennsylvania rifle", plus a "long barreled" after "this" and before "rifled". If anyone has any information on it not being so, I can change it later.

**Comment by I-am-Erk** [comment]
Make sure to update the migrations in Generic Guns before this is merged.

**Comment by Ker-Nes** [comment]
> Make sure to update the migrations in Generic Guns before this is merged.

The Generic Guns mod don't actually touch in any way the flintlock weapons that I can see, and primitive guns are already "genericized", even with this new addition.

**Pull request merged**